### PR TITLE
[Fix #12547] User experience for infinite loop detected

### LINKED ISSUE #12547: User experience for "Infinite loop detected"
If a user encounters an "Infinite loop detected" error, then I think it makes sense for RuboCop to provide some actionable advice, e.g.:

1. Suggest to the user to update to the latest RuboCop version, if not already in use.
2. Suggest to report a bug if it still occurs on the latest version.

### PR DESCRIPTION
**Pull request by Hiroto-Iizuka**
[#12547](https://github.com/rubocop/rubocop/issues/12547)

This pull request includes the following:
- Added a comment suggesting to upgrade Rubocop to the latest version in case of "Infinite loop detected" error.
- Modified RSpec accordingly.

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/runner.rb [hunk 1]
```diff
@@ -20,6 +20,9 @@ def initialize(path, offenses_by_iteration, loop_start: -1)
         message = 'Infinite loop detected'
         message += " in #{path}" if path
         message += " and caused by #{root_cause}" if root_cause
+        message += ' \n\nPlease update to the latest RuboCop version if not already in use,'
+        message += ' \nand report a bug if the issue still occurs on this version.'
+        message += ' \n\nPlease check the latest version at "https://rubygems.org/gems/rubocop"'
```
Comment (andyw8) [inline lib/rubocop/runner.rb:25]: I'd suggest dropping the quotes around the URL.

File: lib/rubocop/runner.rb [hunk 2]
```diff
@@ -20,6 +20,9 @@ def initialize(path, offenses_by_iteration, loop_start: -1)
         message = 'Infinite loop detected'
         message += " in #{path}" if path
         message += " and caused by #{root_cause}" if root_cause
+        message += ' \n\nPlease update to the latest RuboCop version if not already in use,'
```
Comment (andyw8) [inline lib/rubocop/runner.rb:23]: It _might_ be problematic to have newlines here, especially if other programs need to consume the output. The maintainers can probably say more.

**Comment by andyw8** [comment]
Nice!

File: lib/rubocop/runner.rb (hunk 2 above)
Comment (Hiroto-Iizuka) [reply lib/rubocop/runner.rb]: @andyw8 

Thanks for the review!
I rewrote it in a form without line breaks.

**Comment by andyw8** [comment]
@koic any thoughts?

**Comment by bbatsov** [comment]
I think that's a good small improvement. Sorry for taking so long to merge this!

**Pull request merged**